crates/bevy_ecs/src/query/iter.rs

```diff
@@ -1094,12 +1114,27 @@ impl<'w, 's, D: QueryData, F: QueryFilter> Iterator for QueryIter<'w, 's, D, F>
             accum = func(accum, item);
         }
 
-        for id in self.cursor.storage_id_iter.clone().copied() {
-            // SAFETY:
-            // - The range(None) is equivalent to [0, storage.entity_count)
-            accum = unsafe { self.fold_over_storage_range(accum, &mut func, id, None) };
+        // We're going to consume the iterator, and don't want to mutably borrow self twice, so
```

Comment:
I considered/tried doing that, but that just drew my attention to the fact that the current (and previous) fold implementations cloned the iterator - but that seems surprising to me. I'm not sure if I'm being dumb, but I would've expected a fold call to _consume_ the iterators, and when I realised that I thought this might be "more correct".

What are your thoughts on that?

Reply:
I just remembered that fold takes `self` by value, not ref, so it doesn't matter - the clone can't cause externally observable behaviour differences (e.g. you can't call fold twice and get repeated elements).

Reply:
I've made this change locally, and will push it soon, so marking this resolved.